frontend/apps/reader/modules/readerview.lua

```diff
@@ -815,11 +815,29 @@ end
 
 function ReaderView:onSetRotationMode(rotation)
     if rotation ~= nil then
-        if rotation == Screen:getRotationMode() then
+        if rotation == Screen:getRotationMode() then -- no change
             return true
         end
+
+        local requested_screen_mode -- landscape or portrait
+        if rotation == Screen.DEVICE_ROTATED_UPRIGHT or rotation == Screen.ICE_ROTATED_UPSIDE_DOWN then
+            requested_screen_mode = "portrait"
+        else
+            requested_screen_mode = "landscape"
+        end
+
+        local old_rotation_mode = Screen:getRotationMode()
+        local old_screen_mode = Screen:getScreenMode()
+        if rotation and rotation ~= old_rotation_mode and requested_screen_mode == old_screen_mode then
```

Comment:
The first nil guard is unnecessary, you're already in a `if rotation ~= nil` branch